### PR TITLE
Add support to dpy2

### PR DESCRIPTION
**Pull request by ltzmax**
This is a drafted pr where i'll add support for some expect those 2 prs thats already open to add support for it.

### CODE REVIEW

**Comment by ltzmax** [comment]
Alright, should be ready now. 
The only i didnt do: steamcommunity, reverseimagesearch and captcha
- Steamcommunity require to support current lib version which it doesn't, either lock to a version where it works or add support for current.
- reverseimagesearch couldnt get api to test it.
- captcha require some work.

File: datautils/__init__.py [hunk 1]
```diff
@@ -17,8 +17,8 @@ async def setup_after_ready(bot):
...
                 if bot.get_command(alias):
                     command.aliases[command.aliases.index(alias)] = f"du{alias}"
-    bot.add_cog(cog)
+    await bot.add_cog(cog)
 
 
-def setup(bot):
-    create_task(setup_after_ready(bot))
+async def setup(bot):
+    await create_task(setup_after_ready(bot))
```
Comment (fixator10) [inline datautils/__init__.py]: [create_task](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task) is sync function
```suggestion
    create_task(setup_after_ready(bot))
```

File: generalchannel/__init__.py [hunk 2]
```diff
@@ -5,5 +5,6 @@
 )
 
 
-def setup(bot):
-    bot.add_cog(GeneralChannel(bot))
+async def setup(bot):
+    cog = GeneralChannel(bot)
+    await bot.add_cog(cog)
```
Comment (fixator10) [inline generalchannel/__init__.py]: Why adding an variable here?
```suggestion
async def setup(bot):
    await bot.add_cog(GeneralChannel(bot))
```

File: godvilledata/__init__.py [hunk 3]
```diff
@@ -7,4 +7,5 @@
 
 
 async def setup(bot):
-    await bot.add_cog(GodvilleData(bot))
+    cog = GodvilleData(bot)
+    await bot.add_cog(cog)
```
Comment (fixator10) [inline godvilledata/__init__.py]: Same here
```suggestion
    await bot.add_cog(GodvilleData(bot))
```

File: massthings/__init__.py [hunk 4]
```diff
@@ -6,5 +6,6 @@
 )
 
 
-def setup(bot):
-    bot.add_cog(MassThings(bot))
+async def setup(bot):
+    cog = MassThings(bot)
+    await bot.add_cog(cog)
```
Comment (fixator10) [inline massthings/__init__.py]:
```suggestion
async def setup(bot):
    await bot.add_cog(MassThings(bot))
```

File: moreutils/__init__.py [hunk 5]
```diff
@@ -17,8 +17,8 @@ async def setup_after_ready(bot):
...
                 if bot.get_command(alias):
                     command.aliases[command.aliases.index(alias)] = f"mu{alias}"
-    bot.add_cog(cog)
+    await bot.add_cog(cog)
 
 
-def setup(bot):
-    create_task(setup_after_ready(bot))
+async def setup(bot):
+    await create_task(setup_after_ready(bot))
```
Comment (fixator10) [inline moreutils/__init__.py]: [create_task](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task) is sync function
```suggestion
    create_task(setup_after_ready(bot))
```

**Review by fixator10 — CHANGES_REQUESTED** [review]

**Comment by ltzmax** [comment]
also note for `weather`, the api is apple's now and will no longer allow you to make api + the current api you have seems to be deleted by apple so i was not able to test the cog properly.

**Comment by fixator10** [comment]
Ye, thats known, i most likely will need to rewrite it entirely

**Review by fixator10 — APPROVED** [review]
Datautils failing, but its not init issue, taking it from here

**Pull request merged**